### PR TITLE
Fix dracut

### PR DESCRIPTION
**Pull request by PhilippHomann**
see #62 

Currently tested on Fedora 28 Workstation for my special use case with TPM2.
Not tested against tang or http, because of missing environment.

### CODE REVIEW

File: src/dracut/module-setup.sh.in [hunk 1]
```diff
@@ -23,18 +23,9 @@ depends() {
     return 0
 }
 
-cmdline() {
-    echo "rd.neednet=1"
```
Comment (npmccallum) [inline src/dracut/module-setup.sh.in]: This change will break network-based policies.

File: src/dracut/module-setup.sh.in [hunk 2]
```diff
@@ -23,18 +23,9 @@ depends() {
...
-}
-
 install() {
     local ret=0
 
-    cmdline > "${initdir}/etc/cmdline.d/99clevis.conf"
```
Comment (npmccallum) [inline src/dracut/module-setup.sh.in]: We still need this.

File: src/dracut/module-setup.sh.in [hunk 3]
```diff
@@ -64,6 +56,27 @@ install() {
...
+Before=basic.target shutdown.target
+EOF
+    done
+
+    # remove all dependencies etc from original files
+    sed -i '/^[Before|Conflicts|Requires|After|Wants].*/d' \
+        "${unitdir}/clevis-luks-askpass.service" \
+	"${unitdir}/clevis-luks-askpass.path"
+
+    #
```
Comment (npmccallum) [inline src/dracut/module-setup.sh.in]: Stray comment.

File: src/dracut/module-setup.sh.in [hunk 4]
```diff
@@ -64,6 +56,27 @@ install() {
...
+    for f in path service; do
+        cat > "${unitdir}/clevis-luks-askpass.${f}.d/00-dracut.conf" <<EOF
+[Unit]
+DefaultDependencies=no
+Conflicts=shutdown.target
+Before=basic.target shutdown.target
+EOF
+    done
+
+    # remove all dependencies etc from original files
```
Comment (npmccallum) [inline src/dracut/module-setup.sh.in]: This is a complete sentence. Please capitalize it and terminate it with a period.

File: src/dracut/module-setup.sh.in [hunk 5]
```diff
@@ -46,7 +37,8 @@ install() {
         mktemp \
         curl \
         jose \
-        nc
+        nc \
+        grep
```
Comment (npmccallum) [inline src/dracut/module-setup.sh.in]: Why is grep needed?

Comment (PhilippHomann) [reply src/dracut/module-setup.sh.in]: grep is needed by /usr/libexec/clevis-luks-askpass.
If you mind the overhead of grep I can also change the script to use bash internals.

File: src/dracut/module-setup.sh.in (hunk 1 above)
Comment (PhilippHomann) [reply src/dracut/module-setup.sh.in]: I think using rd.neednet is not the correct way when having a systemd service.
That way I removed these entries.

Normally we have to rely on network-online.target, but this does not work for me.
In my testing environment I added Wants and After for clevis-luks-askpass.path, but it starts before the interface has an IP address set.

So the only thing I can imagine to do is having a hacky workaroung by checking if any interface has an IP.
Or do you have any other suggestion about that?

**Comment by PhilippHomann** [comment]
I think I will take care of your points 1 and 2 when you agreed with all other changes.
So see my recent comments and changes.

By the way: have you ever seen clevis running in combination with dracut?
Because as mentioned in #62 for me the execution of clevis was blocked by the password prompt.

**Comment by PhilippHomann** [comment]
Okay for whatever reason everything is working now in my lab.
Sorry for disturbing you and wasting your time.

**Pull request closed without merging**